exonum/src/crypto/mod.rs

```diff
@@ -851,59 +851,64 @@ impl CryptoHash for Decimal {
 
 #[cfg(test)]
 mod tests {
-    use super::{gen_keypair, hash, Hash, HashStream, PublicKey, SecretKey, Seed, SignStream,
-                Signature, EMPTY_SLICE_HASH};
-    use encoding::serialize::FromHex;
+    use super::*;
+
     use serde_json;
+    use serde::de::DeserializeOwned;
+
+    use encoding::serialize::FromHex;
 
     #[test]
-    fn test_hash() {
-        let h = hash(&[]);
-        let h1 = Hash::from_hex(h.to_hex()).unwrap();
-        assert_eq!(h1, h);
-        let h = Hash::zero();
-        assert_eq!(*h.as_ref(), [0; 32]);
+    fn to_from_hex_hash() {
+        let original = hash(&[]);
+        let from_hex = Hash::from_hex(original.to_hex()).unwrap();
+        assert_eq!(original, from_hex);
     }
 
     #[test]
-    fn test_keys() {
+    fn zero_hash() {
+        let hash = Hash::zero();
+        assert_eq!(hash.as_ref(), [0; 32]);
+    }
+
+    #[test]
+    fn to_from_hex_keys() {
         let (p, s) = gen_keypair();
-        let p1 = PublicKey::from_hex(p.to_hex()).unwrap();
-        let s1 = SecretKey::from_hex(s.to_hex()).unwrap();
-        assert_eq!(p1, p);
-        assert_eq!(s1, s);
+
+        let ph = PublicKey::from_hex(p.to_hex()).unwrap();
+        assert_eq!(p, ph);
+
+        let sh = SecretKey::from_hex(s.to_hex()).unwrap();
+        assert_eq!(s, sh);
     }
 
     #[test]
-    fn test_serialize_deserialize() {
-        let h = Hash::new([207; 32]);
-        let json_h = serde_json::to_string(&h).unwrap();
-        let h1 = serde_json::from_str(&json_h).unwrap();
-        assert_eq!(h, h1);
+    fn serialize_deserialize_hash() {
+        assert_serialize_deserialize(&Hash::new([207; 32]));
```

Comment:
Question: as far as I can see, all the tested hashes are filled with the same value (`0xCF` here). 
Do we have (and do we need) any tests to check that byte order is correct? As far as I can see, current tests will pass if somehow serialization will be done in little-endian, and deserialization in big-endian.

Reply:
I don't want to rewrite tests in this pull request, but it is worth creating a separate issue.